kedro/framework/cli/pipeline.py

```diff
@@ -942,6 +942,7 @@ def _generate_setup_file(
             "config/**/parameters*",
             "config/parameters*/**",
             "config/parameters*/**/*",
+            "config/parameters*/**/**/*",
```

Comment:
It looks like actually [recursive globs aren't supported by setuptools](https://github.com/pypa/setuptools/issues/1806), so even though we use the `**` syntax it doesn't do anything different from just the single `*`.  "The patterns here are glob-style patterns: * matches zero or more regular filename characters (on Unix, everything except forward slash; on Windows, everything except backslash and colon);" https://packaging.python.org/guides/using-manifest-in/

Reply:
Ahhh that is very annoying 😬 I'm a bit surprised our tests were already passing if `**` doesn't work in setup.py at all, but I guess we must have not been testing that case.

Looking at the code I think it would be best if we stopped using globs in `_generate_setup_file` altogether and just use `configs_to_package = _find_config_files` as the single source of truth. Looks like `conf_target` should already have all the right parameters files in it, so all we should need to do is list recursively _all_ (no need for any pattern matching) the files in that directory, pass that into `_generate_setup_file` and add `README.md` to the list?

Would be good to get @lorenabalan's thoughts on this one.

Reply:
Yeah I agree with the above, if `setuptools` is not doing what we want, let's compute the list of config files ourselves and just pass them to the `setup.py` file.

Reply:
🤦

Reply:
I updated the implementation as discussed.